bot/exts/halloween/spooknamerate.py

```
@@ -0,0 +1,329 @@
+        """Tell the current random word (name)."""
+        if not self.poll:
+            return await ctx.send(f"The name is **{self.word}**")
+        else:
+            return await ctx.send(f"The word ~~is~~ was **{self.word}**. The poll has already started, so you cannot \
+add an entry.")
+
+    @spook_name_rate.command(name="add", aliases=["+", "register"])
+    async def add_word(self, ctx: Context, *, word: str) -> None:
+        """A command that adds your word!"""
+        if not self.poll:
+            message = ctx.message  # get the message
+
+            for message_id in self.messages:
+                data = self.messages.get(message_id)
+                if data['author'] == message.author:  # if the author has already added an entry
+                    return await ctx.send(f"But you have already added an entry! Type `{self.bot.command_prefix}spooknamerate \
+delete` to delete it, and then you can add it again")
+                elif data['word'] == word:
+                    # Someone has added the word
+                    return await ctx.send("TOO LATE. Someone has already added this word.")
+
+            # otherwise
+            self.messages[message.id] = {  # store it
+                'word': word,  # store the word
+                'author': message.author,  # and the author
+                'score': 0,  # and the score
+            }
+
+            for emoji in self.emojis_val.keys():  # get the emojis
+                await message.add_reaction(emoji)  # and add them
+
+            logger.info(f"{message.author} added the word {word!r}")
+            # display if entry was successful
+            return await ctx.send(f"{word!r} added successfully!\n{random.choice(self.added_messages)}")
+
+        else:
+
+            logger.info(f"{ctx.message.author} tried to add a word, but the poll had already started.")
+            await ctx.send("Sorry, the poll has started! You can try and participate in the next round though!")
```

I'd rather put this at the top with a `if self.poll`, and fail fast with a return after. That way we removed a whole indenation level of the command logic.

---

Did that, and it looks much better. Thank you!